EIPS/eip-2569.md

```diff
@@ -0,0 +1,357 @@
+
+The existing solutions still keep such an image in a centralized server instead of Ethereum. When these applications display an image for a token they retrieve the token's information from Ethereum and search the centralized server for the token's associated image by using the token's information.
+
+Although this is an applicable way to display an image for a token, the image is still vulnerable to risks of being damaged or lost when saved in a centralized server.
+
+Hence we propose a set of interfaces to save an image for a universal token in Ethereum to keep the image permanent and tamper-resistant, and to retrieve an image for a universal token from Ethereum. 
+
+## Definitions
+```tokenId```: for a non-fungible token such as an ERC-721 token or a multi-token such as an ERC-1155 token which has a member "ID" to specify its token type or token index our proposed interface assigns an SVG image's file content to a string variable of the token's contract and  associates the SVG image to this "ID" number. This unique ID is used to access its SVG image in both a "set" operation and a "get" operation. 
+
+For a fungible token such as an ERC-20 token no such an ID is needed and our proposed interface just assigns an SVG image's file content to a string variable of the token's contract.
+
+```imageSvg```: we use a string variable to save an SVG image file's content.
+
+An SVG image that will be saved in the imageSvg string should include at least two attributes:"name", "desc"(description).
+
+```setTokenImageSvg```: for an ERC-721 or ERC-1155 token or a token implemented by a contract which has a member "ID" to specify its token type or token index we define an interface to associate an SVG image to the token's ID number. For an ERC-20 token or a token implemented by a contract which doesn't have a member "ID" to specify its token type or token index we define an interface to assign an SVG image to a member variable string of this token's contract.
+
+```getTokenImageSvg```: for an ERC-721 or ERC-1155 token or a token implemented by a contract which has a member "ID" to specify its token type or token index we define an interface to get an SVG image by using the token's ID number. For an ERC-20 token or a token implemented by a contract which doesn't have a member "ID" to specify its token type or token index we define an interface to get an SVG image for it if the token has a member variable string to save the image.
+
+
+## Specification
+Our proposed interfaces define how a smart contract saves an image for a universal token in Ethereum which keeps the image permanent and tamper-resistant, and how a smart contract retrieves an image from Ethereum for a universal token.  
+
+By calling our proposed interfaces we assume users are accessing an SVG image. 
```

Comment:
When authoring an ERC, write it as a specification rather than a suggestion of a specification.  Instead of saying "Our proposed interfaces define..." say something like "An EIP-2569 compatible contract **MUST** have a method with the signature `getTokenImageSvg(uint256) view returns (string memory)`" or similar.

Reply:
Changes made and committed